### PR TITLE
Add/keep python3.8 support (dict -> Dict) for generated code

### PR DESCRIPTION
**Pull request by gst**
Fix: https://github.com/mirumee/ariadne-codegen/issues/187

### CODE REVIEW

File: ariadne_codegen/client_generators/client.py [hunk 1]
```diff
@@ -262,7 +263,7 @@ def _generate_variables_assign(
         return generate_ann_assign(
             target=self._variables_dict_variable,
             annotation=generate_subscript(
-                generate_name("dict"),
+                generate_name("Dict"),
```
Comment (mat-sop) [inline ariadne_codegen/client_generators/client.py:266]: We should use the constant as well
```suggestion
                generate_name(DICT),
```

File: tests/client_generators/test_client_generator.py [hunk 2]
```diff
@@ -295,7 +297,7 @@ def test_add_method_generates_correct_async_method_body(
         ast.AnnAssign(
             target=ast.Name(id="variables"),
             annotation=ast.Subscript(
-                value=ast.Name(id="dict"),
+                value=ast.Name(id="Dict"),
```
Comment (mat-sop) [inline tests/client_generators/test_client_generator.py:300]:
```suggestion
                value=ast.Name(id=DICT),
```

File: tests/client_generators/test_client_generator.py [hunk 3]
```diff
@@ -449,7 +451,7 @@ def test_add_method_generates_correct_method_body(
...
```
Comment (mat-sop) [inline tests/client_generators/test_client_generator.py:454]:
```suggestion
                value=ast.Name(id=DICT),
```

File: tests/client_generators/test_client_generator.py [hunk 4]
```diff
@@ -545,7 +547,7 @@ def test_add_method_generates_async_generator_for_subscription_definition(
             ast.AnnAssign(
                 target=ast.Name(id="variables"),
                 annotation=ast.Subscript(
-                    value=ast.Name(id="dict"),
+                    value=ast.Name(id="Dict"),
```
Comment (mat-sop) [inline tests/client_generators/test_client_generator.py:550]:
```suggestion
                    value=ast.Name(id=DICT),
```

File: ariadne_codegen/graphql_schema_generators/utils.py [hunk 5]
```diff
@@ -13,11 +13,19 @@
...
+        return "Tuple"
+    return obj.__class__.__name__
+
+
 def get_named_type(type_: GraphQLNamedType, type_map_name: str) -> ast.Call:
     return generate_call(
         func=generate_name("cast"),
         args=[
-            generate_name(type_.__class__.__name__),
+            generate_name(_keep_dict_tuple_capitalized(type_)),
```
Comment (mat-sop) [inline ariadne_codegen/graphql_schema_generators/utils.py:28]: Hmm, I don't see a point in this change. `type_` has the annotation `GraphQLNamedType`, so it shouldn't get `dict` or `tuple`. Am I missing something? Do you see a case where it's necessary?

**Review by mat-sop — CHANGES_REQUESTED** [review]

**Comment by gst** [comment]
hmm @mat-sop 

ideally this would not deserve a python3.8 build/CI (run) test configuration too ?

**Review by mat-sop — APPROVED** [review]

**Comment by mat-sop** [comment]
> hmm @mat-sop
> 
> ideally this would not deserve a python3.8 build/CI (run) test configuration too ?

Ideally yes, but most of our tests check if the generated package looks like expected and we can't run those tests on `3.8`
To include `3.8` in our ci we would need to:
1. Use separate python versions for generation and client tests.
2. Create tests that check only the behavior of the generated client.

`1.` seems relatively easy to do, but `2.` definitely requires more work, so it may be something to consider in the future. Of course, if you want to create those tests, contributions are very welcome 😄

**Comment by gst** [comment]
> Use separate python versions for generation and client tests.

yes obviously :)

> Create tests that check only the behavior of the generated client.

that would be out of the scope of this PR ;)

but you could have simple checks, kind of : 

1) import every generated sub-module, must succeed, obviously.
2) egrep "[^\w](dict|tuple)\[" **/*.py ; for not top-level annotations.

eventually.

**Review by rafalp — APPROVED** [review]
LGTM. We'll need to remember to mention this change in changelog after this is merged.

**Pull request merged**